### DISCUSSION TITLE
Looks like rendering issue with surface plot

### DISCUSSION THREAD

**Discussion by DaMilyutin** [plotters-rs/plotters#299] [Q&A]
Hi,

I'm playting with sixtyfps. I've changed plotter example https://github.com/sixtyfpsui/sixtyfps/tree/master/examples/plotter to plot gauss surface and have plotting depending on az and el (i.e. pitch and yaw).
On some particular values of el and az happens something like this
![image](https://user-images.githubusercontent.com/44639272/139559220-a265ab3d-8210-4619-b0af-e1587d361cce.png)
![image](https://user-images.githubusercontent.com/44639272/139559233-4143c587-b45a-4368-98df-6a1b454fe706.png)

And this one is ok
![image](https://user-images.githubusercontent.com/44639272/139559242-639c56eb-b846-4d06-9886-5e5c3b46a3a8.png)

Code for plotting
``` // what I've changed in plotting example
fn pdf(x: f64, y: f64) -> f64 {
    const SDX: f64 = 0.1;
    const SDY: f64 = 0.05;
    const A: f64 = 5.0;
    let x = x as f64 / 10.0;
    let y = y as f64 / 10.0;
    A * (-x * x / 2.0 / SDX / SDX - y * y / 2.0 / SDY / SDY).exp()
}

fn render_plot(el: f32, az: f32) -> sixtyfps::Image {
    let mut pixel_buffer = SharedPixelBuffer::new(640, 480);
    let size = (pixel_buffer.width() as u32, pixel_buffer.height() as u32);

    let backend = BitMapBackend::with_buffer(pixel_buffer.make_mut_bytes(), size);

    // Plotters requires TrueType fonts from the file system to draw axis text - we skip that for
    // WASM for now.
    #[cfg(target_arch = "wasm32")]
    let backend = wasm_backend::BackendWithoutText { backend };

    let root = backend.into_drawing_area();

    root.fill(&WHITE).expect("error filling drawing area");

    let mut chart = ChartBuilder::on(&root)
        .build_cartesian_3d(-3.0..3.0, 0.0..6.0, -3.0..3.0)
        .expect("error building coordinate system");
    chart.with_projection(|mut p| {
        p.pitch = 0.017453292 * el as f64;
        p.yaw = 0.017453292 * az as f64;
        p.scale = 0.7;
        p.into_matrix() // build the projection matrix
    });

    chart.configure_axes().draw().expect("error drawing");

    let scale = 10. as f64;
    let ind = (3. * scale) as i64;

    chart
        .draw_series(
            SurfaceSeries::xoz(
                (-ind..=ind).map(|x| x as f64 / scale),
                (-ind..=ind).map(|x| x as f64 / scale),
                pdf,
            )
            .style_func(&|&v| {
                (&HSLColor(240.0 / 360.0 - 240.0 / 360.0 * v / 5.0, 1.0, 0.7)).into()
            }),
        )
        .expect("error drawing series");

    root.present().expect("error presenting");
    drop(chart);
    drop(root);

    sixtyfps::Image::from_rgb8(pixel_buffer)
}
```

With this and angle you probably reproduce these effects.

I'm not sure if all is done right on my side.
So give me advise if you see any misses.

Feel free to request any additional data or code.

**Comment by DaMilyutin** [comment]
@38 , sorry to bother. Do you need any additional information?
"sixtyfps" is now "slint" and one should update example to make it more obvious.
I suppose one can reproduce this without sixtyfps/slint if right el,az(point of view/rotation matrix) are chosen.

**Reply by 38** [reply]
Actually, I just realized the gtk-demo also has some issue with the surface plot. Could you please verify that this is the same issue you are talking about, that would be very helpful.

**Reply by DaMilyutin** [reply]
@38, I'm not sure what gtk-demo is.

Seems to me you can do a little modification in your demo
https://github.com/plotters-rs/plotters/blob/master/plotters/examples/3d-plot2.rs
to rotate it in heading. And then effect will be seen. 

If you want to I'll prepare code for you when I have some spare time.

I could actually make demo more interesting. Like both pitch+heading scenario and pdf to have 2 different sigmas.
What do you think about that?
